### PR TITLE
feat: add template page

### PR DESCRIPTION
**Pull request by yusualhashash**

### CODE REVIEW

**Review by a0m0rajab — CHANGES_REQUESTED** [review]
Please update the UI within this PR, and add filtersBy flag to the tables, 

image

and make sure to have the same alignment for buttons: 

image

**Comment by yusualhashash** [comment]
done can you check it now after update  ui

File: apps/web/src/app/[lang]/app/[type]/template/rebate/page.tsx [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+    }
+
+    if (filteredSetupData.length > 0) {
+      payload.setupData = filteredSetupData;
+    }
+  };
+
+  const isRowEmpty = (row: any): boolean => {
+    return Object.values(row).every((value) => value === "" || value === null);
+  };
```
Comment (a0m0rajab) [inline apps/web/src/app/[lang]/app/[type]/template/rebate/page.tsx:256]: We should enable this function.

**Review by a0m0rajab — APPROVED** [review]
Approve in favor of updating it later: 
- the submit function should be functional (at lasting printing json data to the console) 
- the button should be aligned with the table.  
image

**Comment by a0m0rajab** [comment]
@yusualhashash can you fix the lint issues?

**Pull request merged**